Fix Nullpointer for Gradle Parsing

sconvent opened the pull request:
This will fix issue #6384 

The problem was that Dependabot initially recognized correctly that there is no `build.gradle.kts` in the root folder, but later tried to access it anyways (See `top_level_build_file` and the comment above [here](https://github.com/dependabot/dependabot-core/compare/main...sconvent:dependabot-core:fix-nullpointer-gradle#diff-529cac3df5acb2177d1528ec5190f5c8a9e0c59d2d4bf094b5b014ebaab72cadR86))

This problem is solved with an additional `.compact()` statement, which will remove any `nil` entries in the list of files to be looked into.

Tested it by executing it on (an old state) of one of my repos with:
`bin/dry-run.rb gradle sconvent/dependabot-issue-demonstrator --commit addf67d1aa51dade24e47d1043abbd0baa54332a`

Works as expected (i.e. no Nullpointers)

It will probably also fix #4793, as it is a very similar issue. 

Happy to get some feedback as this is my first PR for Dependabot.

jeffwidman commented on gradle/lib/dependabot/gradle/file_parser/property_value_finder.rb:
```diff
@@ -86,7 +86,7 @@ def property_details(property_name:, callsite_buildfile:)
             FileParser.find_includes(callsite_buildfile, dependency_files),
             FileParser.find_includes(top_level_buildfile, dependency_files)
-          )
+          ).compact()
```
I checked, and `FileParser.find_includes()` already protects against hitting NullPointer:
https://github.com/dependabot/dependabot-core/blob/5dc907e3148961c74ddc0b4494730f8e86118f79/gradle/lib/dependabot/gradle/file_parser.rb#L51

So this should be the only code change needed 👍

jeffwidman reviewed (commented):
LGTM, however could you add a unit test for this scenario to prevent future regressions?

Odds of regressing on this exact problem are low, but the fact that we're hitting an error here shows we don't have any test coverage for this scenario of a missing top-level file... and odds of regressing in some other way for this scenario are high enough that we ought to get that test in place.

It should be straightforward to write by copying one of the existing tests, but if you run into questions / need help please let me know.

jeffwidman commented:
Also, lint issue needs fixing, it's trivial 😁

jeffwidman commented:
Actually there may be a unit test in https://github.com/dependabot/dependabot-core/pull/4256 I am on my phone so can't look in detail.

jeffwidman commented on gradle/lib/dependabot/gradle/file_parser/property_value_finder.rb:
```diff
@@ -86,7 +86,7 @@ def property_details(property_name:, callsite_buildfile:)
           all_files = [callsite_buildfile, top_level_buildfile].concat(
             FileParser.find_includes(callsite_buildfile, dependency_files),
             FileParser.find_includes(top_level_buildfile, dependency_files)
```
I checked, and `FileParser.find_includes()` already protects against hitting NullPointer:
https://github.com/dependabot/dependabot-core/blob/5dc907e3148961c74ddc0b4494730f8e86118f79/gradle/lib/dependabot/gradle/file_parser.rb#L51

sconvent commented:
Linting issue should be fixed now.

Sure, I'll add a testcase. Thanks for the hint where I can find one.

jeffwidman commented:
Thanks. Feel free to squash your commits unless they're logically separate, or I'm also happy to do that at merge time. We aren't sticklers for single commit PR's esp if multiple discrete changes need to land in a single PR, but no need for a distinct commit just to fix a lint issue.

sconvent commented:
Added a testcase (for each Groovy and Kotln DSL) and squashed everything into one commit.
The testcases are a bit different from the one you mentioned. @jeffwidman . But they failed without the changes in the actual code and now succeed.

jeffwidman reviewed (approved):
Thank you! Tests look reasonable to me.
